app/uk/gov/hmrc/alcoholdutyreturns/controllers/ReturnsController.scala

```diff
@@ -56,16 +58,34 @@ class ReturnsController @Inject() (
   def submitReturn(appaId: String, periodKey: String): Action[JsValue] =
     authorise(parse.json).async { implicit request =>
       withJsonBody[AdrReturnSubmission] { returnSubmission =>
-        returnsService
-          .submitReturn(returnSubmission, ReturnId(appaId, periodKey))
-          .map(AdrReturnCreatedDetails.fromReturnCreatedDetails)
-          .fold(
-            e => {
-              logger.warn(s"Unable to submit return $periodKey for $appaId: $e")
-              error(e)
-            },
-            returnCreatedDetails => Created(Json.toJson(returnCreatedDetails))
-          )
+        ReturnPeriod
+          .fromPeriodKey(periodKey)
+          .fold {
+            logger.info(s"Invalid period key $periodKey when submitting return")
+            Future.successful(error(ErrorResponse.BadRequest))
+          } { returnPeriod =>
+            returnSubmission
+              .validate(returnPeriod)
+              .fold(
+                failures => {
+                  logger.info(
+                    s"Validation errors when submitting return: ${failures.toChain.toList.map(_.errorMessage).mkString("; ")}"
```

Comment:
Is info okay for this one, or should it be warn ?

Reply:
Or error for that matter. Something has gone wrong which shouldn't happen.

Reply:
Warn should be fine here. It might be affecting only one user which doesn't need a 'waking up in the middle of the night' :D